addons/web/static/src/js/views/form_widgets.js

```diff
@@ -1590,6 +1591,62 @@ var FieldToggleBoolean = common.AbstractField.extend({
     },
 });
 
+/**
+    This widget is intended to be used on Text fields. It will provide Ace Editor for editing XMLs.
+*/
+var AceEditor = common.AbstractField.extend(common.ReinitializeFieldMixin, {
+    template: "AceEditor",
+    init: function() {
+        this._super.apply(this, arguments);
+        ace_call.load();
```

Comment:
ace_call.load() load both mode xml and python, we need only one mode at time so we can pass option and load only require mode